qt/python/mantidqt/utils/qt/qappthreadcall.py

```diff
@@ -47,7 +48,10 @@ def __call__(self, *args, **kwargs):
         else:
             self._ensure_self_on_qapp_thread()
             self._store_function_args(args, kwargs)
-            QMetaObject.invokeMethod(self, "on_call", Qt.BlockingQueuedConnection)
+            if self.blocking:
```

Comment:
It can be good practice to avoid repetition of the function/method call itself in cases like this and instead just adjust the parameter so that when reading the code it's obvious that only a parameter differs. In this case it's not too bad but for more parameters you need to more carefully read both clauses to see if the other arguments are the same, e.g.
```python
            connection_type = Qt.BlockingQueuedConnection if self.blocking else Qt.AutoConnection
            QMetaObject.invokeMethod(self, "on_call", connection_type)
```